pkg/filters/calico/calico.go

```
@@ -0,0 +1,246 @@
+			}
+			output = strings.Trim(output, "\n")
+			if output == expectedOutput {
+				for j := 0; j <= 10; j++ {
+					run = fmt.Sprintf("iptables --list | grep %s", entry)
+					iptablesOutput, err := runCmd(run)
+					if err != nil {
+						return false, fmt.Errorf("executing command %q: %w", run, err)
+					}
+					lines := strings.Count(iptablesOutput, entry)
+
+					if lines == expectedNumberOfMatchesInIptables {
+						return true, nil
+					}
+
+					time.Sleep(1 * time.Second)
+				}
+				return true, nil
+			}
+		}
+
+		// Wait for one second before checking again.
+		time.Sleep(1 * time.Second)
+	}
+
+	return false, nil
+}
+
+func (b *calicoCNI) CleanUp() {
+	gnpManifest, err := util.RenderTemplate(globalNetworkPolicyTmpl, b)
+	if err != nil {
+		fmt.Printf("rendering GlobalNetworkPolicy template: %w", err)
+	}
+
+	gnpWorkloadsManifest, err := util.RenderTemplate(gnpTmplForWorkloads, b)
+	if err != nil {
+		fmt.Printf("rendering GlobalNetworkPolicy workloads template: %w", err)
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		fmt.Printf("creating cluster config: %w", err)
+	}
+
+	for _, gnsManifest := range b.GlobalNetworkSetManifests {
+		// Decode and apply the GlobalNetworkSet manifest
+		if err := util.DecodeAndApply(config, gnsManifest, "DELETE"); err != nil {
+			fmt.Printf("deleting GlobalNetworkSets: %w", err)
+		}
+	}
+	// Decode and apply the GlobalNetworkPolicy manifest
+	if err := util.DecodeAndApply(config, gnpManifest, "DELETE"); err != nil {
+		fmt.Printf("deleting GlobalNetworkPolicy: %w", err)
+	}
+	// Decode and apply the GlobalNetworkPolicy manifest
+	if err := util.DecodeAndApply(config, gnpWorkloadsManifest, "DELETE"); err != nil {
+		fmt.Printf("deleting GlobalNetworkPolicy: %w", err)
+	}
+}
+
+func runCmd(run string) (string, error) {
+	cmd := exec.Command("sh", "-c", run)
+	// Ignoring the err here as it returns `exit status 1`
+	// Actual message is captured in output.
```

I think it would be better to return the error here and then ignore it in `waitUntilReady`. Then, `waitUntilReady` could also ignore the returned error from `listIpsets`.

This is because `runCmd` is also used in `setUp` where I guess the error should not be ignored. Is it correct?